pkg/admissioncontroller/webhooks/admission/auditpolicy/admission.go

```diff
@@ -0,0 +1,294 @@
+	var (
+		oldCm = &corev1.ConfigMap{}
+		cm    = &corev1.ConfigMap{}
+	)
+	if err := h.getOldObject(request, oldCm); err != nil {
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+
+	if request.Operation == admissionv1.Delete {
+		if controllerutil.ContainsFinalizer(oldCm, shoot.FinalizerName) {
+			return admission.Errored(http.StatusUnprocessableEntity, fmt.Errorf("deletion of audit-policy is forbidden as it is still referenced by a shoot (cm has finalizer)"))
+		}
+		return acadmission.Allowed("deletion of audit-policy accepted")
+	}
+
+	if request.Operation == admissionv1.Update {
+		if err := h.decoder.Decode(request, cm); err != nil {
+			return admission.Errored(http.StatusInternalServerError, err)
+		}
+		if !controllerutil.ContainsFinalizer(cm, shoot.FinalizerName) {
+			return acadmission.Allowed("configmap is not referenced by a Shoot")
+		}
+
+		auditPolicy, err := getAuditPolicy(cm)
+		if err != nil {
+			return admission.Errored(http.StatusUnprocessableEntity, err)
+		}
+
+		oldAuditPolicy, ok := oldCm.Data[auditPolicyConfigMapDataKey]
+		if ok {
+			if oldAuditPolicy == auditPolicy {
+				return acadmission.Allowed("audit policy did not change")
+			}
+		}
+
+		// Validate audit policy semantically
+		auditPolicyObj, schemaVersion, err := decoder.Decode([]byte(auditPolicy), nil, nil)
+		if err != nil {
+			return admission.Errored(http.StatusUnprocessableEntity, fmt.Errorf("failed to decode the provided audit policy err=%v", err))
+		}
+		auditPolicyInternal, ok := auditPolicyObj.(*audit_internal.Policy)
+		if !ok {
+			return admission.Errored(http.StatusInternalServerError, fmt.Errorf("failure to cast to audit Policy type: %v", schemaVersion))
+		}
+		errList := auditvalidation.ValidatePolicy(auditPolicyInternal)
+		if len(errList) != 0 {
+			return admission.Errored(http.StatusUnprocessableEntity, fmt.Errorf("provided invalid audit policy err=%v", errList))
+		}
+
+		// Validate schema version against Shoot Kubernetes version
+		existingShoots := &gardencorev1beta1.ShootList{}
+		if err := h.apiReader.List(ctx, existingShoots, client.InNamespace(request.Namespace)); err != nil {
+			return admission.Errored(http.StatusInternalServerError, err)
+		}
+		if err := validatePolicyAgainstShoots(request, schemaVersion, existingShoots); err != nil {
+			return admission.Errored(http.StatusUnprocessableEntity, err)
+		}
+
+		return acadmission.Allowed("configmap change is valid")
+	}
+	return acadmission.Allowed("operation is not update or delete")
+}
+
+// IsValidAuditPolicyVersion checks whether the api server support the provided audit policy apiVersion
+func IsValidAuditPolicyVersion(shootVersion string, schemaVersion *schema.GroupVersionKind) (bool, error) {
+	auditGroupVersion := schemaVersion.GroupVersion().String()
+
+	if auditGroupVersion == "audit.k8s.io/v1" {
+		return version.CheckVersionMeetsConstraint(shootVersion, ">= v1.12")
+	}
+	return true, nil
+}
```

Comment:
You can add a TODO to update this check when https://github.com/kubernetes/kubernetes/issues/98035 is resolved.